### PR TITLE
Adds some Shrine updates from Material team.

### PR DESCRIPTION
**Pull request by RedBrogdon**
@willlarche, you may recall the changes you helped me with just prior to #AskFlutter. This PR adds them to the actual repo for posterity

### CODE REVIEW

File: shrine/lib/login.dart [hunk 1]
```diff
@@ -58,32 +58,36 @@ class _LoginPageState extends State<LoginPage> {
...
               child: TextField(
                 controller: _passwordController,
-                decoration: InputDecoration(
+                decoration: const InputDecoration(
                   labelText: 'Password',
                 ),
               ),
             ),
-            ButtonBar(
+            Wrap(
```
Comment (willlarche) [inline shrine/lib/login.dart]: What does this change do?

Comment (RedBrogdon) [reply shrine/lib/login.dart]: Hmm. This was probably introduced for the flutter/flutter version of Shrine, which has an extra button (exit).  It's not serving much of a purpose here, other than keeping the code in sync with the other copy.

I can take it out.

Comment (willlarche) [reply shrine/lib/login.dart]: That exit button don't look good anyway. Should be a back button at the top.

Comment (RedBrogdon) [reply shrine/lib/login.dart]: In its defense, it was hacked into the code at the last minute by an engineer with no eye for the visual. 😄

**Pull request merged**